build-system-abstraction.rst

```diff
@@ -0,0 +1,472 @@
+    A mandatory key, this is a list of Python format strings [#strformat]_
+    describing the command to run. For instance, if using flit then the build
+    command might be::
+
+        build_command: ["flit"]
+
+    If using a command which is a runnable module fred::
+
+        build_command: ["{PYTHON}", "-m", "fred"]
+
+Process interface
+-----------------
+
+The command to run is defined by a simple Python format string [#strformat]_.
+
+This permits build systems with dedicated scripts and those that are invoked
+using "python -m somemodule".
+
+Processes will be run with the current working directory set to the root of
+the source tree.
+
+When run, processes should not read from stdin - while pip currently runs
+build systems with stdin connected to it's own stdin, stdout and stderr are
+redirected and no communication with the user is possible.
+
+As usual with processes, a non-zero exit status indicates an error.
+
+Available format variables
+--------------------------
+
+PYTHON
+    The Python interpreter in use. This is important to enable calling things
+    which are just Python entry points.
+
+        {PYTHON} -m foo
+
+Available environment variables
+-------------------------------
+
+These variables are set by the caller of the build system and will always be
+available.
+
+PATH
+    The standard system path.
+
+PYTHON
+    As for format variables.
+
+PYTHONPATH
+    Used to control sys.path per the normal Python mechanisms.
+
+Subcommands
+-----------
+
+There are a number of separate subcommands that build systems must support.
+The examples below use a build_command of ``flit`` for illustrative purposes.
+
+build_requires
+    Query build requirements. Build requirements are returned as a UTF-8
+    encoded JSON document with one key ``build_requires`` consisting of a list
+    of dependency specifications [#dependencyspec]_. Additional keys must be
+    ignored. The build_requires command is the only command run without
+    setting up a build environment.
+
+    Example command::
+
+        flit build_requires
+
+metadata
+    Query project metadata.  The metadata and only the metadata should
+    be output on stdout in UTF-8 encoding. pip would run metadata just once to
+    determine what other packages need to be downloaded and installed. The
+    metadata is output as a wheel METADATA file per PEP-427 [#pep427]_.
+
+    Note that the metadata generated by the metadata command, and the metadata
+    present in a generated wheel must be identical.
+
+    Example command::
+
+        flit metadata
+
+wheel -d OUTPUT_DIR
+    Command to run to build a wheel of the project. OUTPUT_DIR will point to
+    an existing directory where the wheel should be output. Stdout and stderr
+    have no semantic meaning. Only one file should be output - if more are
+    output then pip would pick an arbitrary one to consume.
+
+    Example command::
+
+        flit wheel -d /tmp/pip-build_1234
+
+develop [--prefix PREFIX]
```

Comment:
But translate it to what? There aren't enough variables being passed on to the subprocess to represent the necessary information in any form.

I don't know if the osx_framework_user thing is documented anywhere, but it's an install scheme defined in the code:
https://hg.python.org/cpython/file/tip/Lib/sysconfig.py#l72